### ISSUE TITLE
TrainCarts blocks spigot server query

### ISSUE THREAD

**Issue by KenDerGuru** [bergerhealer/TrainCarts#42]
I already posted this at a wrong place, so here it is now at GitHub:

Hello mg_1999,
I really like your plugin and it got essential to my server's structure.
I found a bug, which needs to be fixed:
The plugin TrainCarts, not only BKCommonLib, blocks every query communication of the server with outside. For example if I want to access the online players with the control panel of my hoster, it says "0 players online "and "server is offline". This happens as well, when you want to make a banner like in the signatures of many forum participiants. There is the server query blocked as well.
If I remove TrainCarts from my plugins folder, everything works fine. Reseting all the plugin files doesn't change anything. I am using the newest versions of TrainCarts, BKCommonLib and I even just created a absolutely new spigot file with BuildTools.
I hope you can fix this and I wish you a happy new year 2016!

P.S: I would really appreciate it, if you could make this plugin premium, but therefore you do more updates or bug fixes, because I really need TrainCarts for my server to work.

**Comment by Bart-0110** [comment]
The server is offline is also a bug in MC itself. The first load it can't seem to find some of the servers.

**Comment by KenDerGuru** [comment]
That's not what I mean. The minecraft server in the Minecraft/Multiplayer-Serverlist in the client is shown normally and works perfectly well, but other queries like the webinterface of my hoster doesn't recognize the server as offline and can't get information from it. As well some services like Voting-Sites or other important server sites are not able to fetch data from the server, AND ONLY, when TrainCarts is enabled and even if I use the default configuration, so the bug is caused by TrainCarts.
It would be nice, if the developer could find the problem for this and fix it. Thanks.

**Comment by mg-1999** [comment]
Traincarts doesnt interfere with any network connections related to ping.
So it would be some kind of thing in BKC that is causing it. I know that
the tool on dinnerbone.com is working for my servers and probably the
voting website is doing some shady shit that isnt supported.

Op ma 4 jan. 2016 18:47 schreef KenDerGuru <notifications@github.com>:

> That's not what I mean. The minecraft server in the
> Minecraft/Multiplayer-Serverlist in the client is shown normally and works
> perfectly well, but other queries like the webinterface of my hoster
> doesn't recognize the server as offline and can't get information from it.
> As well some services like Voting-Sites or other important server sites are
> not able to fetch data from the server, AND ONLY, when TrainCarts is
> enabled and even if I use the default configuration, so the bug is caused
> by TrainCarts.
> It would be nice, if the developer could find the problem for this and fix
> it. Thanks.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bergerhealer/TrainCarts/issues/42#issuecomment-168749480>
> .
>
-- 

Met vriendelijk groeten,

Matthijs Giezenberg

**Comment by KenDerGuru** [comment]
But when I only have BKCommonLib in my plugins folder, it works. When I put TrainCarts to it it gets broken. Do you have any idea, what could be the problem for this? Some class in TrainCarts, which is years old and obsolete? It would be really nice if you could find this.

**Comment by mg-1999** [comment]
Not every part of BKC is active when it is the only plugin, there are some
base classes in it that are used by TC and then activated which is probably
causing the trouble. Can you check if the server is providing the correct
information on https://dinnerbone.com/minecraft/tools/status/ . It would
help to identify the problem.

Op di 5 jan. 2016 00:35 schreef KenDerGuru <notifications@github.com>:

> But when I only have BKCommonLib in my plugins folder, it works. When I
> put TrainCarts to it it gets broken. Do you have any idea, what could be
> the problem for this? Some class in TrainCarts, which is years old and
> obsolete? It would be really nice if you could find this.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bergerhealer/TrainCarts/issues/42#issuecomment-168844577>
> .
>
-- 

Met vriendelijk groeten,

Matthijs Giezenberg

**Comment by mg-1999** [comment]
Somebody who tested it? And still got the error?

**Comment by KenDerGuru** [comment]
At dinnerbone server status website everything gets displayed correctly with no issues even the player count. And I reinstalled Citizens shortly and for 2 days or so my server webinterface showes everything correctly. After 2 days it broke again. I moved traincarts out of the folder and it worked. 

I really dont know whats wrong, but I think my hoster uses for its webinterface some old code with interfereces with something of multiple plugins, and as long it is no "real" issue" for my server, I think this thread is closed. If I find any point WHY this randomly happens I will maybe wirte a ticket again. Thanks for your support.
